packages/smart-vaults/src/vault/SmartVault.sol

```diff
@@ -38,48 +38,23 @@ contract SmartVault is IAccount, Ownable, UUPSUpgradeable, LightSyncMultiSigner,
         bytes data;
     }
 
-    /**
-     * @notice Primary Signature types
-     * @dev UserOp signature type is also used in ERC1271 signature verification flow with the assumption that the
-     * userOp Signature is of type Single.
-     */
-    enum SignatureType {
-        UserOp,
-        LightSync
-    }
-
-    /// @notice Primary signature
-    struct Signature {
-        SignatureType sigType;
-        /**
-         * @dev Signature can be of the following types:
-         *      UserOp: abi.encode(UserOpSignature)
-         *      LightSync: abi.encode(LightSyncSignature)
-         */
-        bytes signature;
+    /// @notice Primary Signature types
+    enum SignatureTypes {
+        SingleUserOp,
+        MerkelizedUserOp,
+        LightSyncSingleUserOpSignature,
+        LightSyncMerkelizedUserOpSignature,
+        ERC1271,
+        LightSyncERC1271
```

Comment:
I think this could be improved. I would try to have a single `LightSync` signature type that can be paired with any of the other signature types. Something like this:

- Rename `LightSyncSignature` &rarr; `SignedSignerSetUpdate` to avoid conflict with the following.
- Replace all the `LightSync*` enum members with a single `LightSync`.
- Replace all the `LightSync*Signature` structs with a single `LightSyncSignature` struct. The type of `signature` should be identified via `_getSignatureType` like a "top-level" signature and processed recursively.
```solidity
    struct LightSyncSignature {
        LightSyncSignature[] lightSyncSignatures;
        bytes signature;
    }
```

Reply:
shouldn't the LightSyncSignature[] inside LightSyncSignature be SignedSignerSetUpdate?

Reply:
Yes, sorry! Fixed.